### ISSUE TITLE
Glib Error when using the library with python3-gi

### ISSUE THREAD

**Issue by nekatak** [snapcore/snapd-glib#88]
On ubuntu-18.04 and having installed with apt-get these packages: `libsnapd-glib-dev libsnapd-glib1 snapd` I am able to use some of the snapd client functions (like installing/removing packages) but it is failing to get configuration for a snap.

An example script is:
```
from gi.repository import Snapd

client = Snapd.Client()
client.connect_sync()
print(client.get_snap_conf_sync('nextcloud', None, None))
```
After you install the `nextcloud` (or any other that actually has configuration) running the  above will make python crash with the following error:

```
GLib:ERROR:../../../../glib/gvarianttypeinfo.c:184:g_variant_type_info_check: assertion failed: (0 <= index && index < 24)
Aborted
```
If you try to get configuration for a snap that doesnt have config then you get an empty dictionary in response.

Any help would be appreciated.

**Comment by robert-ancell** [comment]
Thanks for finding this @nekatak! The introspection annotations were wrong for the returned value, which caused a double free. With the fix in #89 I can run your Python program to get snap config.

Note that `SnapdClient.connect_sync` is no-longer required. The client will connect automatically.

**Comment by robert-ancell** [comment]
Fixed in 1.58 and done as a [stable release update](https://bugs.launchpad.net/ubuntu/+source/snapd-glib/+bug/1903936).
